core/src/main/java/com/hamdan/forzenbook/core/UtilFunctions.kt

```diff
@@ -33,24 +36,38 @@ fun getBitmapFromUri(
     uri: Uri,
     context: Context,
     onError: () -> Unit,
-    onBitmapLoaded: (Bitmap) -> Unit
+    onBitmapLoaded: (Bitmap) -> Unit,
+    onImageSaved: (File?) -> Unit
 ) {
     Glide
         .with(context)
         .asBitmap()
         .error(R.drawable.baseline_square_24)
         .load(uri)
-        .into(BitmapGetter(onError = onError) { onBitmapLoaded(it) })
+        .into(object : SimpleTarget<Bitmap>() {
+            override fun onResourceReady(resource: Bitmap, transition: Transition<in Bitmap>?) {
+                // Save the downloaded bitmap to a temporary file
+                var tempFile: File? = null
+                try {
+                    tempFile = createImageFile(context)
+                    FileOutputStream(tempFile).use { fos ->
+                        resource.compress(Bitmap.CompressFormat.JPEG, 100, fos)
+                    }
+                } catch (e: IOException) {
+                    e.printStackTrace()
+                    onError()
+                }
+                onBitmapLoaded(resource)
+                onImageSaved(tempFile)
+            }
+        })
 }
 
-class BitmapGetter(val onError: () -> Unit, val onGetBitmap: (Bitmap) -> Unit) :
-    CustomTarget<Bitmap>() {
-    override fun onLoadCleared(placeholder: Drawable?) {}
-    override fun onResourceReady(resource: Bitmap, transition: Transition<in Bitmap>?) {
-        try {
-            onGetBitmap(resource)
-        } catch (e: Exception) {
-            onError()
-        }
+fun createImageFile(context: Context): File? {
+    val storageDir: File? = context.getExternalFilesDir(Environment.DIRECTORY_PICTURES)
```

Comment:
You're creating a temp file to a non temp file location. This isn't a temp cache directory

Reply:
yea I wasn't sure if I had to specify, updated it after rereading the documentation for temp file, if I dont supply a directory it makes it in the temporary directory.